testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/securityapi/TestCustomAbstractServlet.java

```diff
@@ -0,0 +1,99 @@
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.HttpMethodConstraint;
+import jakarta.servlet.annotation.ServletSecurity;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+/**
+ * A simple servlet to test using the custom method of {@link TestCustomPrincipal}, for Jakarta annotations
+ * {@link Inject} and {@link Resource}.
+ *
+ * @author <a href="mailto:carodrig@redhat.com">Cameron Rodriguez</a>
+ */
+public abstract class TestCustomAbstractServlet extends HttpServlet {
+
+    private SecurityContext securityContext = null;
+
+    static final String LOGIN_HEADER = "Login-Time";
+
+    abstract SecurityContext setSecurityContext();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        SecurityContext securityContext = setSecurityContext();
+
+        try (PrintWriter pw = resp.getWriter()) {
+            final Principal principal = Objects.requireNonNull(securityContext, "SecurityContext must be injected")
+                    .getCallerPrincipal();
+             pw.println("<html><head><title>doGet - TestCustomPrincipal</title></head>");
+            pw.println("<body><h1>Custom principal test, via SecurityContext</h1>");
+
+            if (principal instanceof TestCustomPrincipal) {
+                TestCustomPrincipal customPrincipal = (TestCustomPrincipal) principal;
+
+                pw.printf("<ul>Principal: %s</li>", customPrincipal.getName());
+                pw.printf("<li>Class: %s</li>\n", customPrincipal.getClass().getCanonicalName());
+                pw.printf("<li>Current login time: %s</li>\n", customPrincipal.getCurrentLoginTime().toString());
+                pw.println("</ul></body></html>");
+
+                resp.addHeader(LOGIN_HEADER, customPrincipal.getCurrentLoginTime().format(ISO_LOCAL_DATE_TIME));
+            } else {
+                pw.printf("<p>The principal returned was not of class TestCustomPrincipal.</p>");
+                pw.printf("<ul>Principal: %s</li>", principal.getName());
+                pw.printf("<li>Class: %s</li>\n", principal.getClass().getCanonicalName());
+                pw.println("</ul></body></html>");
+
+                resp.setStatus(500);
+            }
+
+        }
+    }
+}
+
+@WebServlet(urlPatterns = "/inject")
+class TestCustomInjectServlet extends TestCustomAbstractServlet {
+    private static final long serialVersionUID = -6856625542729631383L;
+    @Inject
+    private SecurityContext securityContext;
+
+    @Override
+    SecurityContext setSecurityContext() {
+        return securityContext;
+    }
+}
+
+/** Secured Servlets using Elytron authentication require an extra annotation */
+
+@ServletSecurity(httpMethodConstraints = { @HttpMethodConstraint(value = "GET", rolesAllowed = { "Login" }) })
```

Comment:
@cam-rod Just a minor, I think this annotation can be put above TestCustomInjectServlet to make it simpler and have only 1 servlet class? Like this?

```
@WebServlet(urlPatterns = "/inject")
@ServletSecurity(httpMethodConstraints = { @HttpMethodConstraint(value = "GET", rolesAllowed = { "Login" }) })
class TestCustomInjectServlet extends TestCustomAbstractServlet {
    private static final long serialVersionUID = -6856625542729631383L;
    @Inject
    private SecurityContext securityContext;

    @Override
    SecurityContext setSecurityContext() {
        return securityContext;
    }
}
```

Reply:
yeah, this was a leftover from testing both resource and inject, will simplify it